crates/ra_ide/src/completion/complete_unqualified_path.rs

```diff
@@ -38,7 +38,15 @@ fn complete_enum_variants(acc: &mut Completions, ctx: &CompletionContext) {
     if let Some(ty) = ctx.expected_type_of(&ctx.token.parent()) {
         if let Some(Adt::Enum(enum_data)) = ty.as_adt() {
             let variants = enum_data.variants(ctx.db);
-            let module = enum_data.module(ctx.db);
+
+            let module = if let Some(module) = ctx.scope().module() {
+                // Compute path from the completion site if available.
+                module
+            } else {
+                // Otherwise fall back to the enum's definition site.
+                enum_data.module(ctx.db)
```

Comment:
Not sure if this is useful, maybe we should just not add suggestions in this case?

Reply:
I think this could be useful in theory, as that'll be juse `Enum::Variant`. I think, in practice, this'll be a dead code, but I see no harm keeping it

bors r+

Reply:
Also, be careful with master branch right now, #4127 broke EVERYTHING, the fix is on CI rn